### PR TITLE
使用 PyTorch 在 MNIST 数据集上进行逻辑回归

### LINKED ISSUE #5670: 使用 PyTorch 在 MNIST 数据集上进行逻辑回归
* 原文链接：[Logistic Regression on MNIST with PyTorch](https://towardsdatascience.com/logistic-regression-on-mnist-with-pytorch-b048327f8d19)
* Markdown文件：[文件地址](https://github.com/xitu/gold-miner/blob/master/TODO1/logistic-regression-on-mnist-with-pytorch.md)
* PR 地址：#5673 
* 文章分类：**AI**
* **注意：文件位置在 `TODO1` 文件夹中，不是之前的 `TODO` 了。**
----
* 翻译时间：2 天
* 校对时间：1 天
* 翻译积分：2 分
* 校对积分：0.5-1 分
* [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by lsvih**
译文翻译完成，resolve #5670

### CODE REVIEW

**Comment by fireairforce** [comment]
校对认领

**Comment by fanyijihua** [comment]
@fireairforce 好的呢 :beer:

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
-**Logistic regression** is used to describe data and to explain the relationship between **one dependent binary variable** and one or more nominal, ordinal, interval or ratio-level independent variables[1]. The figure below shows the difference between **Logistic** and **Linear** regression.
+**逻辑回归（Logistic Regression）**既可以用来描述数据，也可以用来解释数据中各个二值变量、类别变量、顺序变量、距离变量、比率变量之间的关系[1]。下图展示了**逻辑回归**与**线性回归**的区别。
 
 ![Taken from [https://www.sciencedirect.com/topics/nursing-and-health-professions/logistic-regression-analysis](https://www.sciencedirect.com/topics/nursing-and-health-professions/logistic-regression-analysis)](https://cdn-images-1.medium.com/max/2000/1*xFhICZgdr2VEZQ-C4FLUEA.jpeg)
 
-In this post, I’ll show how to code a Logistic Regression Model in PyTorch.
+本文将展示如何使用 PyTorch 编写逻辑回归模型。
 
-We’ll try and solve the classification problem of MNIST dataset. First, let’s import all the libraries we’ll need.
+我们将尝试在 MNIST 数据集上解决分类问题。首先，导入所有我们需要用到的库：
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: [首先，导入所有我们需要用到的库] =>[首先，导入我们所需要的所有库],我觉得这样比较通顺一点

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 2]
```diff
@@ -22,41 +22,41 @@ import torchvision.transforms as transforms
 import torchvision.datasets as dsets
 ```
 
-I prefer to keep the following list of steps in front of me when creating a model. This list is present on the PyTorch website [2].
+我喜欢在创建模型前列这么一个步骤表。PyTorch 官网[2]上也有这个步骤列表：
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: [我喜欢在创建模型前列这么一个步骤表。PyTorch 官网[2]上也有这个步骤列表：] => [在创建模型时，我更倾向于列一个如下的步骤表],这样我觉得比较好

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 3]
```diff
@@ -22,41 +22,41 @@ import torchvision.transforms as transforms
...
-# Step 1. Load Dataset
-# Step 2. Make Dataset Iterable
-# Step 3. Create Model Class
-# Step 4. Instantiate Model Class
-# Step 5. Instantiate Loss Class
-# Step 6. Instantiate Optimizer Class
-# Step 7. Train Model
+```python
+# 第一步：加载数据集
+# 第二部：使数据集可迭代
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: [第二部]=>[第二步]

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 4]
```diff
@@ -123,25 +123,25 @@ for epoch in range(int(epochs)):
...
                 total+= labels.size(0)
-                # for gpu, bring the predicted and labels back to cpu fro python operations to work
+                # 如果用的是 GPU，则要把预测值和标签都取回 CPU，才能用 Python 来计算
                 correct+= (predicted == labels).sum()
             accuracy = 100 * correct/total
             print("Iteration: {}. Loss: {}. Accuracy: {}.".format(iter, loss.item(), accuracy))
 ```
 
-Training, this model for just 3000 iterations gives an **accuracy of 82%**. You can go ahead and tweak the parameters a bit, to see if the accuracy increases or not.
+在训练时，这个模型只需要进行 3000 次迭代就能达到 **82%** 的准确率。你可以试着再调整一下参数，看看还能不能把准确率再调高一点。
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: [你可以试着再调整一下参数] => [你可以试着继续调整一下参数]

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 5]
```diff
@@ -123,25 +123,25 @@ for epoch in range(int(epochs)):
...
 
-A good exercise to get a more deep understanding of Logistic Regression models in PyTorch, would be to apply this to any classification problem you could think of. For Example, You could train a Logistic Regression Model to classify the images of your favorite **Marvel superheroes** (shouldn’t be very hard since half of them are gone :) ).
+如果你想加深对在 PyTorch 中实现逻辑回归的理解，可以把上面的模型应用于任何分类问题。比如，你可以训练一个逻辑回归模型来对你最喜爱的**漫威英雄**的图像做个分类（有一半已经化灰了，所以做分类应该不是很难）~
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: 我觉得这里的这个:)蛮可爱的.可以保留,嘿嘿嘿

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 6]
```diff
@@ -22,41 +22,41 @@ import torchvision.transforms as transforms
...
 ```
 
-So let’s go through these steps one by one.
+下面我们将一步步完成上述的步骤。
 
-### Load Dataset
+### 加载数据集
 
-To load the dataset, we make use of **torchvision.datasets,** a library which has almost all the popular datasets used in Machine Learning. You can check out the complete list of datasets at [3].
+我们使用**torchvision.datasets**来加载数据集。这个库中包含了几乎全部的用于机器学习的流行数据集。在[3]中可以看到完整的数据集列表。
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: [我们使用**torchvision.datasets**来加载数据集。]=>[我们使用 **torchvision.datasets** 来加载数据集。]

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 7]
```diff
@@ -82,31 +82,31 @@ output_dim = 10
...
 
 ```python
 model = LogisticRegression(input_dim, output_dim)
 ```
 
-### Instantiate the Loss Class
+### 实例化 Loss 类
 
-We use the cross-entropy to compute the loss.
+我们将使用交叉熵损失来计算 loss：
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: [我们将使用交叉熵损失来计算 loss] => [我们使用交叉熵损失来计算 loss]

File: TODO1/logistic-regression-on-mnist-with-pytorch.md [hunk 8]
```diff
@@ -22,41 +22,41 @@ import torchvision.transforms as transforms
...
 ```python
 train_loader = torch.utils.data.DataLoader(dataset=train_dataset, batch_size=batch_size, shuffle=True)
 test_loader = torch.utils.data.DataLoader(dataset=test_dataset, batch_size=batch_size, shuffle=False)
 ```
 
-### Create the Model Class
+### 创建模型类
 
-Now, we will create a class that defines the architecture of Logistic Regression.
+现在，我们将创建一个类，用来定义逻辑回归模型的结构：
```
Comment (fireairforce) [inline TODO1/logistic-regression-on-mnist-with-pytorch.md]: 这里翻译成"我们创建一个用来定义逻辑回归模型结构的类"，用做定语感觉比较通顺

**Comment by leviding** [comment]
@lsvih 已经 merge 啦~ 快快麻溜发布到[掘金](https://juejin.im)然后给我发下链接，方便及时添加积分哟。

掘金翻译计划有自己的知乎专栏，你也可以投稿哈，推荐使用一个[好用的插件](https://chrome.google.com/webstore/detail/minimalist-markdown-edito/pghodfjepegmciihfhdipmimghiakcjf?hl=zh-CN)。
专栏地址：https://zhuanlan.zhihu.com/juejinfanyi

**Comment by lsvih** [comment]
@leviding 已分享，地址：https://juejin.im/post/5cc66d946fb9a032286173a7
